### ISSUE TITLE
[BUG] unable to send a message

### ISSUE THREAD

**Issue by seteahconnoc** [nanu-c/axolotl#319]
### Description

I am unable to send a message. (I succeeded once, but I cannot reproduce it.)

### Steps to Reproduce

1. Fresh install of axolotl on a UBports device.
2. Add existing signal contact
3. Send message

**Expected behavior:**
Confirmation of sending process

**Actual behavior:**
get frame and exclamation mark

#### Versions
Axolotl Beta 0.9.4

#### Device
UBports Ubuntu 16.04 (OTA-15) OnePlus bacon

#### Link to Debug Log
As a newbie, I am in need of further assistance on finding this log file. Please, feel free to ask for further details.

**Comment by Flaburgan** [comment]
Signal servers were overloaded since yesterday afternoon. Can you send messages now?

**Comment by JayVii** [comment]
As of this morning, i can still reproduce this. Wasn't able to send a single message. Also registered my account again.

If not everyone is experiencing this, could it also depend on region? I would suspect, a high portion of the new users and hence new signal servers is located in Europe and probably other densely populated areas.

**Comment by Flaburgan** [comment]
All Axolotl contributors are located in Europe. If you are still experiencing the problem then please share the logs using the loggerviewer app (search for textsecure).

**Comment by selfbc** [comment]
Two issues that I have seen with this library after the Signal outage are "+1<redacted> no prekey for contact +1<redacted>, device 1" and error 413, which used to be a sync issue.

**Comment by JayVii** [comment]
unlike @seteahconnoc, I am not using Ubports, but Mobian on the Pinephone. Compiled 0.9.4 via flatpak.

This is some debugging output I gathered, opening the app, writing a message to a contact and waiting for it to fail:
```
mobian@mobian:~$ flatpak run org.nanuc.Axolotl -e qt

time="2021-01-18T15:20:44+01:00" level=info msg="[axolotl] Starting Signal for Ubuntu version 0.9.4"
time="2021-01-18T15:20:44+01:00" level=info msg="[axolotl] Setup completed"
time="2021-01-18T15:20:44+01:00" level=info msg="[axolotl] Axolotl server started"
time="2021-01-18T15:20:44+01:00" level=info msg="[axolotl] Starting axolotl ws"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] Run Backend"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] openDb: /home/mobian/.local/share/textsecure.nanuc/db"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] NewDataStore"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] Using axolotl-web path /app/bin/axolotl-web/dist"
time="2021-01-18T15:20:44+01:00" level=info msg="[axolotl] Axolotl-gui starting for sys: qt"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] NewDataStore finished"
time="2021-01-18T15:20:44+01:00" level=info msg="[axolotl] Loading Chats"
time="2021-01-18T15:20:44+01:00" level=info msg="[axolotl] Db setup finished"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] Start Session after Decryption"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] starting Signal connection"
time="2021-01-18T15:20:44+01:00" level=debug msg="[axolotl] config path:  /home/mobian/.config/textsecure.nanuc"
time="2021/01/18 15:20:44" level=info msg="[axolotl] Registered"
time="2021/01/18 15:20:44" level=debug msg="[axolotl-ws] send request registrationDone"
time="2021/01/18 15:20:44" level=debug msg="[axolotl] refresh contacts 1"
time="2021/01/18 15:20:44" level=debug msg="[textsecure] Websocket Connecting to signal-server"
time="2021/01/18 15:20:45" level=debug msg="[textsecure] Websocket Connected successfully"
time="2021/01/18 15:20:45" level=debug msg="verb:\"PUT\"  path:\"/api/v1/queue/empty\"  headers:\"X-Signal-Timestamp:<REDACTED>\"  id:<REDACTED>"
time="2021/01/18 15:20:45" level=info msg="[textsecure] No new messages"
time="2021/01/18 15:20:45" level=debug msg="[textsecure] Websocket sending message"
time="2021/01/18 15:20:55" level=info msg="[axolotl] Client Connected true"
time="2021/01/18 15:20:55" level=debug msg="[axolotl] send darkmode to client false"
time="2021/01/18 15:20:55" level=debug msg="[axolotl-ws] getRegistrationStatus"
time="2021/01/18 15:20:55" level=debug msg="[axolotl-ws] send request registrationDone"
time="2021/01/18 15:20:56" level=info msg="websocket: close 1001 (going away)"
time="2021/01/18 15:20:57" level=info msg="[axolotl] Client Connected true"
time="2021/01/18 15:20:57" level=debug msg="[axolotl] send darkmode to client false"
time="2021/01/18 15:20:57" level=debug msg="[axolotl-ws] getRegistrationStatus"
time="2021/01/18 15:20:57" level=debug msg="[axolotl-ws] send request registrationDone"
time="2021/01/18 15:20:57" level=error msg="[axolotl-ws] send message websocket: close sent"
time="2021/01/18 15:20:57" level=debug msg="[axolotl-ws] remove client"
time="2021/01/18 15:21:00" level=info msg="[axolotl] Open chat with id:  1"
time="2021/01/18 15:21:00" level=debug msg="[axolotl] Enter chat  +<REDACTED>"
time="2021/01/18 15:21:06" level=debug msg="[axolotl] send message to  +<REDACTED>"
time="2021/01/18 15:21:07" level=debug msg="GET /v2/keys/+<REDACTED>/1 404\n"
time="2021/01/18 15:21:07" level=error msg="[axolotl] send message error status code 404\n 0"
time="2021/01/18 15:21:07" level=info msg="[axolotl] status code 404\n"
time="2021/01/18 15:21:10" level=debug msg="GET /v2/keys/+<REDACTED>/1 404\n"
time="2021/01/18 15:21:10" level=error msg="[axolotl] send message error status code 404\n 0"
time="2021/01/18 15:21:10" level=info msg="[axolotl] status code 404\n"
time="2021/01/18 15:21:13" level=debug msg="[axolotl] SendMessage +<REDACTED> 0"
time="2021/01/18 15:21:13" level=error msg="[axolotl] sending message failed  <REDACTED>"
time="2021/01/18 15:21:13" level=debug msg="[axolotl-ws] UpdateMessageHandlerWithSource  <REDACTED>"
time="2021/01/18 15:21:18" level=info msg="websocket: close 1001 (going away)"
time="2021/01/18 15:21:19" level=info msg="\nout:\n\nerr:\nfile:///app/share/axolotl/guis/qml/qml/MainQML.qml:23: TypeError: Cannot read property 'bottom' of null\nfile:///app/share/axolotl/guis/qml/qml/MainQML.qml:22: TypeError: Cannot read property 'right' of null\nfile:///app/share/axolotl/guis/qml/qml/MainQML.qml:21: TypeError: Cannot read property 'top' of null\nfile:///app/share/axolotl/guis/qml/qml/MainQML.qml:20: TypeError: Cannot read property 'left' of null\n[9:26:0118/152052.934388:ERROR:gles2_cmd_decoder.cc(2843)] [GroupMarkerNotSet(crbug.com/242999)!:C8C208447F000000]GL ERROR :GL_INVALID_ENUM : BackFramebuffer::Create: <- error from previous GL command\n[9:26:0118/152052.966502:ERROR:gles2_cmd_decoder.cc(2843)] [GroupMarkerNotSet(crbug.com/242999)!:E82F17447F000000]GL ERROR :GL_INVALID_ENUM : BackFramebuffer::Create: <- error from previous GL command\n[9:26:0118/152053.534933:ERROR:gles2_cmd_decoder.cc(2843)] [GroupMarkerNotSet(crbug.com/242999)!:D8EA6F447F000000]GL ERROR :GL_INVALID_ENUM : BackFramebuffer::Create: <- error from previous GL command\n\n"
time="2021/01/18 15:21:19" level=info msg="[axolotl] Axolotl-gui finished with error: <nil>"
```

The GL Errors seem unrelated, and do not seem to be an issue at all (they do not appear when `-e server` is used with e.g. firefox). Relevant part appears to be:
```
time="2021/01/18 15:21:00" level=info msg="[axolotl] Open chat with id:  1"
time="2021/01/18 15:21:00" level=debug msg="[axolotl] Enter chat  +<REDACTED>"
time="2021/01/18 15:21:06" level=debug msg="[axolotl] send message to  +<REDACTED>"
time="2021/01/18 15:21:07" level=debug msg="GET /v2/keys/+<REDACTED>/1 404\n"
time="2021/01/18 15:21:07" level=error msg="[axolotl] send message error status code 404\n 0"
time="2021/01/18 15:21:07" level=info msg="[axolotl] status code 404\n"
time="2021/01/18 15:21:10" level=debug msg="GET /v2/keys/+<REDACTED>/1 404\n"
time="2021/01/18 15:21:10" level=error msg="[axolotl] send message error status code 404\n 0"
time="2021/01/18 15:21:10" level=info msg="[axolotl] status code 404\n"
time="2021/01/18 15:21:13" level=debug msg="[axolotl] SendMessage +<REDACTED> 0"
time="2021/01/18 15:21:13" level=error msg="[axolotl] sending message failed  <REDACTED>"
```

I noticed that the telephone number (redacted here) does not contain the country code properly. It would say +123456789 instead of +49123456789. Not sure if that is 'normal' or whether it is causing this issue.

**Comment by nanu-c** [comment]
I think the https://github.com/nanu-c/axolotl/pull/294 solves it

**Comment by JayVii** [comment]
any ETA yet on when this can be merged?

**Comment by Flaburgan** [comment]
I did a first round of review with a few remarks that nanu has to address. But I am not a Go developer so this has limited value.

**Comment by Flaburgan** [comment]
@JayVii 
@nanu-c released a 0.9.5 version which contains the patch, but there are still important problems with it. We're going to publish a 0.9.6 version in the coming days, please wait for it.

**Comment by nanu-c** [comment]
Please open a new issue if this happens again. :)
